packages/@sanity/validation/src/ValidationError.ts

```diff
@@ -0,0 +1,34 @@
+import {
+  Path,
+  ValidationMarker,
+  ValidationError as IValidationError,
+  ValidationErrorOptions,
+  ValidationErrorClass,
+} from '@sanity/types'
+import ExtendableError from 'es6-error'
+
+// Follows the same pattern as Rule and RuleClass. @see Rule
+const ValidationError: ValidationErrorClass = class ValidationError
+  extends ExtendableError
```

Comment:
Related to previous comment - I'm wondering if you have a gut feeling about any unforeseen problems with not extending Error here? I don't think there's all that much value in it (stack trace isn't really relevant for validation errors, etc.). So I think it's worth considering if we can make it not extend from builtin Error.

Reply:
yeah i agree. i liked your suggestion of just relying on structural types so i would take it even further and remove the class too. i originally was thinking maybe someone else would've been relying on instanceof but this class is actually never exported (so it's internal to the repo). I'll give that a try

Reply:
I thought about this a bit more and i came to the conclusion that removing the class isn't super easy https://github.com/sanity-io/sanity/pull/2691#discussion_r692980849

Reply:
That's fine - it was mostly about not extending `Error`. Leaving it as a class for now is fine.